lib/scryptenc/scryptenc.c

```diff
@@ -230,38 +230,35 @@ scryptenc_setup(uint8_t header[96], uint8_t dk[64],
 {
 	uint8_t salt[32];
 	uint8_t hbuf[32];
-	int logN;
 	uint64_t N;
-	uint32_t r;
-	uint32_t p;
 	SHA256_CTX ctx;
 	uint8_t * key_hmac = &dk[32];
 	HMAC_SHA256_CTX hctx;
 	int rc;
 
 	/* Pick values for N, r, p. */
 	if ((rc = pickparams(P->maxmem, P->maxmemfrac, P->maxtime,
-	    &logN, &r, &p, verbose)) != 0)
+	    &P->logN, &P->r, &P->p, verbose)) != 0)
```

Comment:
This changes program behaviour, since the values `logN`, `r`, and `p` are now returned to the caller via the structure `P`.

This is fine, but the functions should be documented as returning the chosen parameters via the structure.